include/app/tabs/camera.hpp

```diff
@@ -1,35 +1,57 @@
-#pragma once
-
-#include <QLabel>
-#include <QMediaPlayer>
-#include <QStackedLayout>
-#include <QString>
-#include <QVideoWidget>
-
-#include "app/config.hpp"
-#include "app/theme.hpp"
-
-class CameraTab : public QWidget {
-    Q_OBJECT
-
-   public:
-    CameraTab(QWidget *parent = nullptr);
-
-   private:
-    QWidget *connect_widget();
-    QWidget *camera_widget();
-    QWidget *input_widget();
-    void connect_stream();
-    void update_status(QMediaPlayer::MediaStatus media_status);
-
-    Theme *theme;
-    Config *config;
-    QLabel *status;
-    QMediaPlayer *player;
-    QString url;
-
-   signals:
-    void connected();
-    void disconnected();
-};
-
+#pragma once
+
+#include <QLabel>
+#include <QMediaPlayer>
+#include <QStackedLayout>
+#include <QString>
+#include <QVideoWidget>
+#include <QComboBox>
+#include <QRadioButton>
+#include <QCamera>
+#include <QCheckBox>
+#include <QCameraViewfinder>
+#include <QCameraViewfinderSettings>
+
+#include "app/config.hpp"
+#include "app/theme.hpp"
+
+class CameraTab : public QWidget {
+    Q_OBJECT
+
+   public:
+    CameraTab(QWidget *parent = nullptr);
+
+   private:
+    QWidget *connect_widget();
+    QWidget *network_camera_widget();
+    QWidget *local_camera_widget();
+    QWidget *local_cam_selector();
+    QPushButton *connect_button();
+    QWidget *network_cam_selector();
+    QWidget *selector_widget(QWidget *selection);
+    bool populate_local_cams();
```

Comment:
why does this return a bool even though its never checked?

Reply:
That's a good question. It used to disable the control if no cams were found. I'll look into it.